assets/app/features/upgrade/controllers/upgrade-landing.controller.spec.js

```diff
@@ -79,6 +211,69 @@ describe('Upgrade Landing Controller', function() {
         should.exist(controller);
     });
 
+    describe('when created while prepare is not running', function() {
+        it('should not start polling for status', function() {
+            expect(upgradeStatusFactory.waitForStepToEnd).not.toHaveBeenCalled();
+        });
+
+        it('prepare should not be running', function() {
+            assert.isFalse(controller.prepare.running);
+        });
+
+        it('prepare should not be completed', function() {
+            assert.isFalse(controller.prepare.completed);
+        });
+    });
+
+    describe('when created while prepare is running', function() {
+        beforeEach(function() {
+            // local change in mocked service
+            upgradeFactory.getStatus =
+                sinon.spy(function () { return $q.when(incompleteUpgradeResponse); });
```

Comment:
Oh, you're right. I did assume the original spy has been set through Jasmine's spy. Now I see your point.
Regarding the library... I like the idea! Let's talk more about it on IRC or hangouts